src/app/clusters/resource-monitoring-server/resource-monitoring-cluster-objects.h

```diff
@@ -141,37 +141,39 @@ struct TypeInfo
 };
 } // namespace LastChangedTime
 
-namespace ReplacementProductStruct {
+namespace GenericReplacementProductStruct {
 enum class Fields : uint8_t
 {
     kProductIdentifierType  = 0,
     kProductIdentifierValue = 1,
 };
 
-struct Type
+class GenericType : protected app::Clusters::HepaFilterMonitoring::Structs::ReplacementProductStruct::Type
 {
-public:
-    ProductIdentifierTypeEnum productIdentifierType = static_cast<ProductIdentifierTypeEnum>(0);
+private:
+    ResourceMonitoring::ProductIdentifierTypeEnum productIdentifierType;
     chip::CharSpan productIdentifierValue;
 
-    CHIP_ERROR Decode(TLV::TLVReader & reader);
-
+public:
+    GenericType() {}
+    GenericType(ResourceMonitoring::ProductIdentifierTypeEnum aProductIdentifierType, chip::CharSpan aProductIdentifierValue)
+    {
+        productIdentifierType  = aProductIdentifierType;
+        productIdentifierValue = aProductIdentifierValue;
+    }
+    virtual ~GenericType()                = default;
     static constexpr bool kIsFabricScoped = false;
-
     CHIP_ERROR Encode(TLV::TLVWriter & writer, TLV::Tag tag) const;
```

Comment:
Why is this not just:
```suggestion
    using HepaFilterMonitoring::Structs::ReplacementProductStruct::Type::Encode;
```
instead of reimplementing things?